### PR TITLE
Introduce typedoc for `@directus/types` for internal purposes

### PR DESCRIPTION
**Pull request by paescuj**
For internal development purposes.

**Usage**
```bash
pnpm typedoc
open docs/index.html
```
Screenshot 2023-05-19 at 22-28-32 @directus_types

### CODE REVIEW

File: packages/types/typedoc.config.cjs [hunk 1]
```diff
@@ -0,0 +1,35 @@
+const fs = require('node:fs');
+const path = require('node:path');
+
+function* walkSync(dir) {
+	const files = fs.readdirSync(dir, { withFileTypes: true });
+
+	for (const file of files) {
+		if (file.isDirectory()) {
+			yield* walkSync(path.join(dir, file.name));
```
Comment (rijkvanzanten) [inline packages/types/typedoc.config.cjs:9]: Any way we can put this in a separate package so we can reuse it in other packages as well? 🤔

Comment (paescuj) [reply packages/types/typedoc.config.cjs:9]: I'm afraid this won't work since we have to use CJS for this config 😢

Comment (paescuj) [reply packages/types/typedoc.config.cjs:9]: Oh you meant the whole config?

**Comment by paescuj** [comment]
Closing for now in favor of a solution which will generate docs for all packages

**Pull request closed without merging**